extensions/widgets/segmented/segmented.lcb

```diff
@@ -946,7 +984,16 @@ private handler setSelectedSegment(in pSelectedSegment as String)
 	variable tNewSelection as List
 	put [] into tNewSelection
 	repeat for each element tSelectedString in tStringList
-		put tSelectedString parsed as number into tSelected
+		if pNames then
+			-- If we're selecting by name, check if the names all exist.
+			put the index of tSelectedString in mSegmentNames into tSelected
+			if tSelected is 0 then
+				throw "segment" && tSelectedString && "does not exist"
+				return
+			end if 
+		else
+			put tSelectedString parsed as number into tSelected
+		end if
```

Comment:
If there are two segments with the same name "foo", and you set the `hilitedItemNames` to "foo,foo", what should happen?